### PR TITLE
[HandshakeToFIRRTL] Implement lowering of StoreOp.

### PR DESCRIPTION
**Pull request by mikeurbach**
The implementation is a simple combinational circuit that waits for
all inputs to be valid and all outputs to be ready.

### CODE REVIEW

File: lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp [hunk 1]
```diff
@@ -1576,6 +1577,62 @@ bool HandshakeBuilder::visitHandshake(MemoryOp op) {
...
+  rewriter.create<ConnectOp>(insertLoc, outputDataValid, allValid);
+  rewriter.create<ConnectOp>(insertLoc, outputAddrValid, allValid);
+
+  // The inputs will be ready when all outputs are ready.
+  auto allReady = rewriter.create<AndPrimOp>(insertLoc, bitType,
+                                             outputDataReady, outputAddrReady);
+
+  rewriter.create<ConnectOp>(insertLoc, inputDataReady, allReady);
+  rewriter.create<ConnectOp>(insertLoc, inputAddrReady, allReady);
+  rewriter.create<ConnectOp>(insertLoc, controlReady, allReady);
```
Comment (hanchenye) [inline lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp:1631]: As the valid/ready logic is similar to JoinOp on the input side, should these ready signals be driven by `allReady && allValid`?

Comment (mikeurbach) [reply lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp:1631]: Shoot good catch. Another reason to have one helper for buildJoinOp! I sent this before that helper was in, so I'll update this to use it.

**Comment by mikeurbach** [comment]
@hanchenye this uses the `buildJoinLogic` helper now. There is a little bit of logic and an extra wire since we need to drive two outputs. As I mentioned [here](https://github.com/llvm/circt/issues/337#issuecomment-758775942), I think we can handle both outputs combinationally without any buffering, but please let me know what you think of this implementation.

**Review by hanchenye — APPROVED** [review]
LGTM.

**Pull request merged**